plugins/ssr/ssrcommon.ml

```diff
@@ -410,7 +410,7 @@ let mk_anon_id t gl_ids =
     let rec loop i j =
       let d = !s.[i] in if not (is_digit d) then i + 1, j else
       loop (i - 1) (if d = '0' then j else i) in
-    let m, j = loop (n - 1) n in m, (!s, j), Id.of_string !s in
+    let m, j = loop (n - 1) n in m, (!s, j), Id.of_string_soft !s in
```

Comment:
Can you add a comment saying why we need to tolerate ill-formed identifiers?

Reply:
I needed to relax this for block ipats when the field lacks a name, iirc. It generates, for a seed x, `_x?_` that is ok but not valid.